packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -132,26 +132,22 @@ export class EthereumAnchorService implements AnchorService {
    */
   private _makeAnchorRequest(carFileReader: AnchorRequestCarFileReader): Observable<CASResponse> {
     return defer(() =>
-      from(
-        this.sendRequest(this.requestsApiEndpoint, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/vnd.ipld.car',
-          },
-          body: carFileReader.carFile.bytes,
-        })
-      )
+      this.sendRequest(this.requestsApiEndpoint, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/vnd.ipld.car',
+        },
+        body: carFileReader.carFile.bytes,
+      })
     ).pipe(
       retry({
         delay: (error) => {
           this._logger.err(
             new Error(
-              `Error connecting to CAS while attempting to anchor ${carFileReader.streamId.toString()} at commit ${carFileReader.tip.toString()}: ${
-                error.message
-              }`
+              `Error connecting to CAS while attempting to anchor ${carFileReader.streamId} at commit ${carFileReader.tip}: ${error.message}`
             )
           )
-          return interval(this.pollInterval)
+          return timer(this.pollInterval)
```

Comment:
UPDATE: Okay I *think* see now that the retry logic will be handled by the `retry` call that is wrapping this, so that's what will make sure the polling keeps happening, this `timer` is just putting a delay between retry attempts.  So I guess it was just wrong that it was using `interval` before.  What is the current behavior here with `interval`?  Does this create an exponential growth of poll calls?  Could this be a potential memory leak?

Reply:
It does not matter. `timer` is just somewhat cleaner. Internal observable (`timer`, or `interval` - does not matter) gets unsubscribed after the first emission.